Samples/Client/SwiftClient/SwiftClient/VideoStreamViewController.swift

```diff
@@ -0,0 +1,640 @@
+    var videoTrack: RTCVideoTrack!
+    var videoStream: RTCMediaStream!
+    var inputChannel: RTCDataChannel!
+    lazy var navTransform: [CGFloat] = {
+        [unowned self] in
+        self.matCreate()
+        }()
+    var navHeading: CGFloat = 0.0
+    var navPitch: CGFloat = 0.0
+    var navLocation: [CGFloat] = [ 0.0, 0.0, 0.0 ]
+    var isFingerDown = false
+    var fingerDownX: CGFloat = 0
+    var fingerDownY: CGFloat = 0
+    var downPitch: CGFloat = 0.0
+    var downHeading: CGFloat = 0.0
+    var downLocation: [CGFloat] = [ 0.0, 0.0, 0.0 ]
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        RTCInitializeSSL()
+        renderView.delegate = self
+        
+        pickerView.dataSource = self
+        pickerView.delegate = self
+        
+        initGestureRecognizer()
+        connect()
+    }
+    
+    func joinPeer(peerId: Int) {
+        createPeerConnection(peerId: peerId)
+        inputChannel = self.peerConnection.dataChannel(forLabel: "inputDataChannel", configuration: RTCDataChannelConfiguration())
+        inputChannel.delegate = self
+        
+        let offerOptions = RTCMediaConstraints(mandatoryConstraints: ["OfferToReceiveAudio": "false","OfferToReceiveVideo": "true"], optionalConstraints: nil)
+        
+        peerConnection.offer(for: offerOptions) { (sdp, error) in
+            let sdpH264 = sdp!.sdp.replacingOccurrences(of: "96 98 100 102", with: "100 96 98 102")
+            let receivedOffer = RTCSessionDescription(type: .offer, sdp: sdpH264)
+            self.peerConnection.setLocalDescription(receivedOffer, completionHandler: { (error) in
+                
+                let jsonData = [
+                    "type": "offer",
+                    "sdp": receivedOffer.sdp
+                ]
+                
+                self.sendToPeer(peerId: peerId, data: self.convertDictionaryToData(dict: jsonData)!)
+            })
+        }
+    }
+    
+    func createPeerConnection(peerId: Int) {
+        newIceCandidatePeerId = peerId
+        
+        let defaultPeerConnectionConstraints = RTCMediaConstraints(mandatoryConstraints: nil,
+                                                                   optionalConstraints: ["DtlsSrtpKeyAgreement": "true"])
+        let rtcConfig = RTCConfiguration()
+        let iceServer = RTCIceServer(urlStrings: ["turn:turnserver3dstreaming.centralus.cloudapp.azure.com:5349"],
+                                     username: "user",
+                                     credential: "3Dtoolkit072017",
+                                     tlsCertPolicy: .insecureNoCheck)
+        rtcConfig.iceServers = [iceServer]
+        rtcConfig.iceTransportPolicy = .relay
+        
+        peerConnection = peerConnectionFactory.peerConnection(with: rtcConfig, constraints: defaultPeerConnectionConstraints, delegate: self)
+    }
+    
+    func handlePan(_ recognizer: UIPanGestureRecognizer) {
+        
```

Comment:
nit: can we remove this space to try and be more consistent? seems in the rest of the codebase you generally don't leave a line before doing work

Reply:
👍